data_discovery_ai/pipeline.py

```diff
@@ -54,20 +66,34 @@ def __init__(
         self.params = params
         self.isDataChanged = isDataChanged
         self.usePretrainedModel = usePretrainedModel
-        self.model_name = model_name
-        self.model = None
-        # TODO: needs to define what are the accepted values
-        #  validate against the list declared in data_discovery_ai/common/constants.py? needs to be a controlled list of values
-        if self.usePretrainedModel and self.model_name is None:
-            raise ValueError("model name should be given to use pretrained model")
+        # validate model name with accepted values, defined in data_discovery_ai/common/constants.py
+        self.validate_model_name(model_name=model_name)
+
+    """
+        Validate model name within fixed selections
+        Input:
+            model_name: str. The file name of the saved model. restricted within four options: development, staging, production, and test
+    """
+
+    def validate_model_name(self, model_name) -> None:
+        valid_model_name = AVAILABLE_MODELS
+        if model_name.lower() not in valid_model_name:
+            raise ValueError(
+                'Available model name: ["development", "staging", "production", "experimental", "benchmark"]'
+            )
+        else:
+            self.model_name = model_name.lower()
 
     def fetch_raw_data(self) -> pd.DataFrame:
         """
         Fetches raw data from Elasticsearch and returns it as a DataFrame.
         Output:
             raw_data: pd.DataFrame. A DataFrame containing the raw data retrieved from Elasticsearch.
         """
-        client = connector.connect_es(config_path="./esManager.config")
+        es_config = configparser.ConfigParser()
+        es_config.read(elasticsearch_config_file_path)
```

Comment:
I think this pattern appearing before, so you can make a method that takes config file path as method's signature, and return config object

e.g read_config(file_path: string) -> config object type

the use it as: es_config = read_config(...path to es config) 

inside this method, you can now avoid duplicated code I mentioned previously.